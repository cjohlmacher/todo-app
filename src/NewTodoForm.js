import React, { useState } from 'react';
import './NewTodoForm.css';
import { useFormik } from 'formik';
import * as yup from 'yup';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import IconButton from '@mui/material/IconButton';
import AddCircleIcon from '@mui/icons-material/AddCircle';

const validationSchema = yup.object({
    todo: yup
        .string('Add a todo')
        .required('Cannot add an empty todo')
});

const NewTodoForm = ({addTodo}) => {

    const formik = useFormik({
        initialValues: {
          todo: ''
        },
        validationSchema: validationSchema,
        onSubmit: (values) => {
            addTodo({description: values.todo});
        },
      });

    return (
        <div>
            <Box sx={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                <form onSubmit={formik.handleSubmit}>
                    <label htmlFor="todo">Add Todo:</label>
                    <TextField
                        id="todo"
                        name="todo"
                        size="small"
                        value={formik.values.todo}
                        onChange={formik.handleChange}
                        error={formik.touched.todo && Boolean(formik.errors.todo)}
                        helperText={formik.touched.todo && formik.errors.todo}
                        sx={{margin: '0px 5px'}}
                    />
                    <IconButton type="submit" aria-label="button">
                        <AddCircleIcon name="add" sx={{color: '#118bd5'}}/>
                    </IconButton>
                </form>
            </Box>
        </div>
        // <Box component="form" sx={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
        //     <label htmlFor="todo">Add Todo:</label>
        //     <TextField id="todo" name="todo" size="small" value={formik.values.todo} onChange={formik.handleChange} sx={{margin: '0px 5px'}}/>
        //     <IconButton type="submit" aria-label="button" onClick={formik.handleSubmit}>
        //         <AddCircleIcon name="add" sx={{color: '#118bd5'}}/>
        //     </IconButton>
        // </Box>
    );
};

export default NewTodoForm;