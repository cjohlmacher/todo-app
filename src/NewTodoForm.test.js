import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('should render without breaking', () => {
    render(<NewTodoForm />);
});