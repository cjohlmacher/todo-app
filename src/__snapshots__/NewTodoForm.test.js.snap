// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-gmuwbf"
    >
      <form>
        <label
          for="todo"
        >
          Add Todo:
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root css-ewmcuh-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="todo"
              name="todo"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-nnbavb"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          aria-label="button"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-92r701-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            name="add"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
